nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -452,82 +432,94 @@ def inference_epoch_end(self, outputs, mode, data_cfg):
 
         averaged_loss = []
         averaged_metric = []
-        metric_name = self.val_metric_name if mode == 'validation' else self.test_metric_name
         # Log metrics for each provided validation/test dataset.
         for dataloader_idx, output in enumerate(outputs):
             loss = super().validation_epoch_end([x['loss'] for x in output])
-            # Determine the key used to log the loss based on the user provided name of the dataset or the dataloader index.
             loss_log_key = self._determine_log_key(data_cfg, dataloader_idx, "loss", mode)
             self.log(loss_log_key, loss)
             averaged_loss.append(loss)
-
-            # Skip the rest of this loop if the user wants to monitor the loss only.
-            if self.val_metric is None:
-                continue
-            # Determine the key used to log the eval metric based on the user provided name of the dataset or the dataloader index.
-            metric_log_key = self._determine_log_key(data_cfg, dataloader_idx, metric_name, mode)
-            metric_object = (
-                self.val_metric[dataloader_idx] if mode == 'validation' else self.test_metric[dataloader_idx]
+            # Gather the outputs object from all data parallel ranks since we are using the DistributedSampler which splits data across DDP ranks.
```

Comment:
is it still possible to skip all this is one just wants to use val loss as the monitor metric?

Reply:
We have this condition to bypass if we use metric.name='loss'
https://github.com/NVIDIA/NeMo/blob/fix-gpt-trainer-test/nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py#L479